File: java_solutions/1379.find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree.java
/*
 * @lc app=leetcode id=1379 lang=java
 *
 * [1379] Find a Corresponding Node of a Binary Tree in a Clone of That Tree
 */

class Solution {
    public final TreeNode getTargetCopy(final TreeNode original, final TreeNode cloned, final TreeNode target) {
        return original == null ? null : (original == target ? cloned : (getTargetCopy(original.left, cloned.left, target) != null ? getTargetCopy(original.left, cloned.left, target) : getTargetCopy(original.right, cloned.right, target)));
    }
}

File: kotlin_solutions_1/1379.find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree.kt
/*
 * @lc app=leetcode id=1379 lang=kotlin
 *
 * [1379] Find a Corresponding Node of a Binary Tree in a Clone of That Tree
 */

class Solution { fun getTargetCopy(original: TreeNode?, cloned: TreeNode?, target: TreeNode?): TreeNode? = if (original == null) null else if (original === target) cloned else getTargetCopy(original.left, cloned?.left, target) ?: getTargetCopy(original.right, cloned?.right, target) }